mlrun/api/crud/pipelines.py

```diff
@@ -371,3 +418,12 @@ def resolve_project_from_workflow_manifest(self, workflow_manifest):
                     raise NotImplementedError(f"Unknown action: {action}")
 
         return mlrun.mlconf.default_project
+
+    @staticmethod
+    def _get_project_from_experiment_name(name: str) -> str:
```

Comment:
I'm not sure where you saw that the experiment name always starts with the project name be if it is indeed the case you can use `startswith` instead of partition.
Also, if you raise an exception here it will fail the whole operation and it won't continue to the next experiment

Reply:
I'm not sure about the `startswith` strategy either. If you have a project called `my-proj` and another called `my-proj-dev`, the experiments for both would be deleted.

About the experiment name pattern, I noticed it here: 
https://github.com/mlrun/mlrun/blob/f91ffc145634d9405d46124759287726bc7454be/mlrun/projects/project.py#L2501

It seems that's also possible for an experiment to only have the project name if the run created has no name of its own. Beyond that, following the call stack that creates runs, it's also technically possible for an experiment to only have the workflow name. See the implementation here: 
https://github.com/mlrun/mlrun/blob/f91ffc145634d9405d46124759287726bc7454be/mlrun/projects/pipelines.py#L583-L583

However, I didn't find an usage of this specific run function that doesn't send name as a parameter, so it's unclear to me if this can really happen on the current codebase. Of course, that says nothing about older versions of MLRun, so my proposed approach is flimsy at best.

I changed my code to better leverage the Kubeflow API and find a stronger connection between Runs and Experiments, so we can avoid this naming problem altogether.